b2sdk/bucket.py

```diff
@@ -111,6 +112,41 @@ def __init__(
         self.is_file_lock_enabled = is_file_lock_enabled
         self.replication = replication
 
+    def _add_file_info_item(self, file_info: dict[str, str], name: str, value: str | None):
+        if value is not None:
+            if name in file_info and file_info[name] != value:
+                logger.warning(
+                    'Overwriting file info key %s with value %s (previous value %s)', name, value,
+                    file_info[name]
+                )
+            file_info[name] = value
+
+    def _merge_file_info_and_headers_params(
+        self,
+        file_info: dict | None,
+        cache_control: str | None,
+        expires: str | dt.datetime | None,
+        content_disposition: str | None,
+        content_encoding: str | None,
+        content_language: str | None,
+    ) -> dict | None:
+        updated_file_info = {**(file_info or {})}
```

Comment:
afaik `file_info` isnt currently nested dictionary, but the nested items would be still modified in both `file_info` and `updated_file_info`, so it's a potential footgun.

Reply:
Hm, `file_info` type hints are inconsistent. I don't think it's worth doing something like a deep copy, fixing `file_info` types to `dict[str, str]` everywhere might be better.